File: 102.cpp
#include <cstdio>
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

int n,x,y,sementara;
unsigned int minimal,temp;
int bin[3][3], list[]= {0,1,2};
char order[] = {'B','C','G'};
bool used[3];
string perm;

void cekbin()
{
	string sekarang;
	int x,y;
	for(x=0;x<3;x++) sekarang+=order[x];
	
	temp=0;
	for(x=0;x<3;x++)
		for(y=0;y<3;y++) if(!(y==list[x])) temp +=bin[x][y];
		
	if((temp<minimal)||((temp==minimal)&&(perm>sekarang)))
	{
		perm=sekarang;
		minimal=temp;
	}
}

void permutasi(int t)
{
	int y;
	if(t=3) cekbin(); else
		for(y=0;y<3;y++) if(used[y])
		{
			list[t]=y;
			used[y]=false;
			permutasi(t+1);
			used[y]=true;
		}
}

int main()
{
	while(cin >> bin[0][0])
	{
		minimal=0;
		
		for(x=0;x<3;x++)
		{
			used[x]=true;
			for(y=0;y<3;y++) if(!((x==0)&&(y==0)))
			{
				cin >> bin[x][y];
				minimal+=bin[x][y];
			}
		}
		
		for(x=0;x<3;x++) swap(bin[x][1],bin[x][2]);
		
		do
		{
			cekbin();
			next_permutation(list,list+3);
		} while(next_permutation(order,order+3));
		
		cout << perm << " " <<minimal<<"\n";
	}
	return 0;
}

File: 102.pas
const kata:array[1..3] of char=('B','G','C');

var x,y:longint;
	temp,min:longword;
	bin:array[1..3,1..3] of longword;
	used:array[1..3] of boolean;
	perm,semen:string;
	list:array[1..3] of longint;
	
procedure cekbin;
	var x,y:longint;
	
	begin
		semen:='';
		for x:=1 to 3 do semen:=semen+kata[list[x]];
		temp:=0;
		for x:=1 to 3 do
			for y:=1 to 3 do if y<>list[x] then inc(temp,bin[x,y]);
		
		if((temp<min)or((temp=min)and(semen<perm))) then
		begin
			perm:=semen;
			min:=temp;
		end;
	end;
	
procedure permutasi(t:longint);
	var y:longint;
	
	begin
		if t=4 then cekbin else
			for y:=1 to 3 do if used[y] then
			begin
				list[t]:=y;
				used[y]:=FALSE;
				permutasi(t+1);
				used[y]:=TRUE;
			end;
	end;
	
begin
	while not(eof) do
	begin
		min:=0;
		for x:=1 to 3 do
		begin
			used[x]:=TRUE;
			for y:=1 to 3 do 
			begin
				read(bin[x,y]);
				inc(min,bin[x,y]);
			end;
		end;
		readln;
		
		perm:='GGG';
		permutasi(1);
		writeln(perm,' ',min);
	end;
end.